### PR TITLE
Adding option to add nvcc functions seperately

### PR DESCRIPTION
**Pull request by cjatin**
-DHIP_NVCC_INLINE_DISABLED to disable inlining of hip wrappers of nvcc functions.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -340,6 +340,19 @@ if(HIP_PLATFORM STREQUAL "hcc")
     file(WRITE "${PROJECT_BINARY_DIR}/.hipInfo" ${_buildInfo})
 endif()
 
+if(HIP_PLATFORM STREQUAL "nvcc")
+    if(NOT DEFINED ENV{CUDA_PATH})
+        set(CUDA_PATH "/usr/local/cuda")
+    else()
+        set(CUDA_PATH $ENV{CUDA_PATH})
+    endif()
```
Comment (emankov) [inline CMakeLists.txt]: Make it platform independent: use UNIX check.

File: CMakeLists.txt [hunk 2]
```diff
@@ -340,6 +340,19 @@ if(HIP_PLATFORM STREQUAL "hcc")
...
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__HIP_PLATFORM_NVCC__")
+    include_directories(${PROJECT_SOURCE_DIR}/include)
+    include_directories(${CUDA_PATH}/include)
+    add_library(hip_nvcc SHARED src/nvcc_functions.cpp)
+    install(TARGETS hip_nvcc DESTINATION lib)
```
Comment (emankov) [inline CMakeLists.txt]: Move install tl install steps.

Comment (cjatin) [reply CMakeLists.txt]: Done

**Comment by mangupta** [comment]
@cjatin Needs minor updates to resolve conflict arising due to PR #1724.

**Pull request closed without merging**